cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/workflow/WorkflowDriver.java

```diff
@@ -81,6 +83,9 @@
   private NettyHttpService httpService;
   private volatile Thread runningThread;
   private final Map<String, WorkflowActionNode> status = new ConcurrentHashMap<String, WorkflowActionNode>();
+  private volatile boolean isSuspended;
```

Comment:
Also, since you always use a lock when accessing the `suspended` variable, it doesn't need to have volatile.